src/cli/commands/install.js

```diff
@@ -494,7 +494,9 @@ export class Install {
     }
 
     // fin!
-    await this.saveLockfileAndIntegrity(topLevelPatterns, workspaceLayout);
+    if (topLevelPatterns.length || await fs.exists(path.join(this.config.cwd, constants.LOCKFILE_FILENAME))) {
+      await this.saveLockfileAndIntegrity(topLevelPatterns, workspaceLayout);
+    }
```

Comment:
One more thing, `fs.exists()` is deprecated since node 8. Should I make a change?
https://nodejs.org/dist/latest-v8.x/docs/api/fs.html#fs_fs_exists_path_callback